ibis/expr/api.py

```diff
@@ -1800,6 +1800,30 @@ def _string_replace(arg, pattern, replacement):
     return ops.StringReplace(arg, pattern, replacement).to_expr()
 
 
+def to_timestamp(arg, format_str, timezone=None):
+    """
+    Parses a string and returns a timestamp.
+
+    Parameters
+    ----------
+    format_str : A format string potentially of the type '%Y-%m-%d'
+    timezone : An optional string indicating the timezone,
+        i.e. 'America/New_York'
```

Comment:
What is the behavior with a non-`None` value of `timezone` with a format string that only includes a date? Does it assume midnight for the time portion of the timestamp?

Reply:
The function assumes midnight for the time component and returns a timestamp that has been converted to UTC. Here's an example:
```
>>> s = ibis.literal('2016-02-22')
>>> expr = s.to_timestamp('%F', 'America/New_York')
>>> client.execute(expr)
Timestamp('2016-02-22 05:00:00')
>>> expr2 = s.to_timestamp('%F', 'UTC')
>>> client.execute(expr)
Timestamp('2016-02-22 00:00:00')
```